### PR TITLE
Invert if should be smart about is_some, is_none, is_ok, is_err

### LINKED ISSUE #5827: Invert if should be smart about is_some/is_none, is_ok, is_err
```rust
if doc_style.is_some() {
                Class::DocComment
            } else {
                Class::Comment
            }
```

->

```rust
if doc_style.is_none() {
                Class::Comment
            } else {
                Class::DocComment
            }
```

Relevant code: 
https://github.com/rust-analyzer/rust-analyzer/blob/844e1aa725cbc26ec8a4221bd5412bd28238544c/crates/assists/src/handlers/invert_if.rs

https://github.com/rust-analyzer/rust-analyzer/blob/844e1aa725cbc26ec8a4221bd5412bd28238544c/crates/assists/src/utils.rs#L268-L270

### PR DESCRIPTION
**Pull request by dragfire**
Fixes #5827

### CODE REVIEW

File: crates/assists/src/utils.rs [hunk 1]
```diff
@@ -179,6 +179,32 @@ fn invert_special_case(expr: &ast::Expr) -> Option<ast::Expr> {
             ast::BinOp::EqualityTest => bin.replace_op(T![!=]).map(|it| it.into()),
             _ => None,
         },
+        ast::Expr::MethodCallExpr(mce) => {
+            let is_some_text = "is_some";
+            let is_none_text = "is_none";
+            let is_ok_text = "is_ok";
+            let is_err_text = "is_err";
```
Comment (bjorn3) [inline crates/assists/src/utils.rs:186]: These can be `const`.

Comment (dragfire) [reply crates/assists/src/utils.rs]: @bjorn3 updated!

File: crates/assists/src/utils.rs [hunk 2]
```diff
@@ -179,6 +179,32 @@ fn invert_special_case(expr: &ast::Expr) -> Option<ast::Expr> {
...
+                || name_text != IS_OK_TEXT
+                || name_text != IS_ERR_TEXT)
+            {
+                return None;
+            }
+
+            let caller = &mce.receiver()?.syntax().text();
+
+            match name_text {
+                x if x == IS_SOME_TEXT => make::expr_method_call(caller, IS_NONE_TEXT),
```
Comment (bjorn3) [inline crates/assists/src/utils.rs:201]: Does `IS_SOME_TEXT => ...` work if you use `match &*name_text`?

File: crates/assists/src/utils.rs [hunk 3]
```diff
@@ -179,6 +179,32 @@ fn invert_special_case(expr: &ast::Expr) -> Option<ast::Expr> {
...
+
+            let name = mce.name_ref()?;
+            let name_text = name.text();
+            if !(name_text != IS_SOME_TEXT
+                || name_text != IS_NONE_TEXT
+                || name_text != IS_OK_TEXT
+                || name_text != IS_ERR_TEXT)
+            {
+                return None;
+            }
```
Comment (bjorn3) [inline crates/assists/src/utils.rs:196]: Can yoy remove this if, and then make `caller` a closure (`|| &mce...`) that you call where you are currently using the `caller` value.

File: crates/assists/src/utils.rs (hunk 2 above)
Comment (dragfire) [reply crates/assists/src/utils.rs:201]: Yes it does. Shall I change it to ```match &*name_text```

**Review by bjorn3 — APPROVED** [review]

**Comment by matklad** [comment]
Thanks!

**Pull request merged**